src/index.js

```diff
@@ -51,13 +60,26 @@ export default function nodeResolve ( options = {} ) {
 
 			return new Promise( ( fulfil, reject ) => {
 				let disregardResult = false;
+				let packageBrowserField = false;
 
 				resolveId(
 					importee,
 					Object.assign({
 						basedir: dirname( importer ),
 						packageFilter ( pkg ) {
-							if ( useModule && pkg[ 'module' ] ) {
+							if (options.browser && typeof pkg[ 'browser' ] === 'object') {
+								packageBrowserField = Object.keys(pkg[ 'browser' ]).reduce((browser, key) => {
+									browser[ key ] = pkg[ 'browser' ][key];
+									if (key[0] === '.' && !extname(key)) browser[ key + '.js'] = browser[ key + '.json' ] = browser[ key ];
+									return browser;
+								}, {});
+							}
```

Comment:
This is a convenient side-effect-ful mechanism of the `packageFilter` function. We know that if we land inside a `package.json` file, that the resolved `importee` has this mapping inside of its own package.json. Therefore, if we collect these `browser` maps inside of a cache (`browserMapCache`) then by the time we get to the `importee` that owns this package.json being the `importer`, all of it's own dependencies can rely on this cache. 

In other words - it is this operation that collects for us, the mapping to which files are replaced in a browser based environment.